Variance's `PrivateItems` assoc const is accessible in subtraits

A [surprising new SemVer breakage case](https://github.com/obi1kenobi/cargo-semver-checks/issues/1658) [was discovered earlier this week](https://github.com/time-rs/time/issues/793), affecting sealed traits' associated items (consts, types, functions with or without receivers — everything) being accessible to subtraits. This is non-obvious and usually isn't intended by the authors of such code, which can lead to accidental breakage.

The standard library [has an at-risk pattern here](https://github.com/rust-lang/rust/blob/7c5dac9c7476a30e09a5aa87b4df2c5663fe7734/library/core/src/marker/variance.rs#L236-L243), where the sealing trait's associated item is accessible downstream: [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=a92c2d31ad9b00a9fae619daa35612fd)
```rust
#![feature(phantom_variance_markers)]

use core::marker::{PhantomCovariant, Variance};

fn leaked_value<T: Variance>() -> T {
    // This is actually `private_items::PrivateItems::VALUE`,
    // which we can't actually name directly.
    T::VALUE
}

fn main() {
    let _: PhantomCovariant<u8> = leaked_value();
}
```

If `Variance` were stabilized, `PrivateItems::VALUE` might also become stabilized. This seems possibly unintentional. Linking the tracking issue to get a backlink here: #135806.

`portable-simd` has the same issue, [filed separately here](https://github.com/rust-lang/portable-simd/issues/537).

Hat tip to @jhpratt for looping me in to look at the SemVer breakage and suggesting that we scan the standard library for this pattern. I used AI tools for the scan, and to generate the code snippet above. I wrote this issue fully by hand otherwise.

Is there a reason that the recommended way of sealing a trait shouldn't now just be to have your public trait depend on a private trait? That used to be banned, but apparently now it's just gated behind a warn-by-default lint:

```rust
mod module {
    trait Sealed {
        fn method(&self);
    }

    #[allow(private_bounds)]
    pub trait Trait: Sealed {}
}

fn foo(x: impl module::Trait) {
    x.method(); // error[E0624]: method `method` is private
}
```

--- a/library/core/src/marker/variance.rs
+++ b/library/core/src/marker/variance.rs
@@ -234,6 +234,7 @@ phantom_type! {
 }
 
 mod private_items {
+    #[unstable(feature = "std_internals", issue = "none")]
     pub trait PrivateItems {
         const VALUE: Self;
     }

